### PR TITLE
Fix handling of frames of an unknown type

### PR DESCRIPTION
**Pull request by DunyaKokoschka**
See: https://datatracker.ietf.org/doc/html/rfc7540#section-4.1

  Implementations MUST ignore and discard any frame that has a
  type that is unknown

### CODE REVIEW

**Review by ericmj — APPROVED** [review]

**Comment by DunyaKokoschka** [comment]
So I had a real site that was sending back these frames but this site no longer sends back these frames so I'm not 100% sure that it fixes the issue. But I do have a conn_test that sends a bad frame and it crashes the test without the fix and doesn't crash with the fix.

File: test/mint/http2/conn_test.exs [hunk 1]
```diff
@@ -18,7 +18,7 @@ defmodule Mint.HTTP2Test do
   setup :start_connection
 
   defmacrop assert_recv_frames([]) do
-    quote do: refute_receive {:ssl, _socket, _data}
+    quote do: refute_receive({:ssl, _socket, _data})
```
Comment (whatyouhide) [inline test/mint/http2/conn_test.exs:21]:
```suggestion
    quote do: refute_receive {:ssl, _socket, _data}
```

File: test/mint/http2/integration_test.exs [hunk 2]
```diff
@@ -46,6 +46,7 @@ defmodule HTTP2.IntegrationTest do
 
   describe "http2.golang.org" do
     @describetag connect: {"http2.golang.org", 443}
+    @describetag skip: "http2.golang.org is down"
```
Comment (whatyouhide) [inline test/mint/http2/integration_test.exs:49]: Let's do this in another PR :)

```suggestion
```

File: test/mint/integration_test.exs [hunk 3]
```diff
@@ -185,6 +185,7 @@ defmodule Mint.IntegrationTest do
       assert merge_body(responses, request) =~ "httpbin.org"
     end
 
+    @describetag skip: "http2.golang.org down"
```
Comment (whatyouhide) [inline test/mint/integration_test.exs:188]: Let's do this in another PR :)

```suggestion
```

File: test/mint/tunnel_proxy_test.exs [hunk 4]
```diff
@@ -83,6 +83,7 @@ defmodule Mint.TunnelProxyTest do
     assert merge_body(responses, request) =~ "httpbin"
   end
 
+  @tag skip: "http2.golang.org down"
```
Comment (whatyouhide) [inline test/mint/tunnel_proxy_test.exs:86]: Let's do this in another PR :)

```suggestion
```

File: test/mint/tunnel_proxy_test.exs [hunk 5]
```diff
@@ -105,6 +106,7 @@ defmodule Mint.TunnelProxyTest do
     assert merge_body(responses, request) =~ "Protocol: HTTP/2.0"
   end
 
+  @tag skip: "http2.golang.org down"
```
Comment (whatyouhide) [inline test/mint/tunnel_proxy_test.exs:109]: Let's do this in another PR :)

```suggestion
```

File: lib/mint/http2.ex [hunk 6]
```diff
@@ -1421,6 +1421,10 @@ defmodule Mint.HTTP2 do
     end
   end
 
+  defp assert_valid_frame(_conn, unknown()) do
+    # we should ignore unknown frames
```
Comment (whatyouhide) [inline lib/mint/http2.ex:1425]:
```suggestion
    # Unknown frames MUST be ignored:
    # https://datatracker.ietf.org/doc/html/rfc7540#section-4.1
    :ok
```

File: test/mint/http2/conn_test.exs [hunk 7]
```diff
@@ -71,6 +71,51 @@ defmodule Mint.HTTP2Test do
     end
   end
 
+  describe "handling unknown frames from the server" do
```
[the remark this answers is not in the document]
Comment (bmteller) [reply test/mint/http2/conn_test.exs:74]: I've cleaned up the formatting.

[the remark this answers is not in the document]
Comment (DunyaKokoschka) [reply test/mint/http2/conn_test.exs:74]: I've cleaned up the formatting.

**Comment by whatyouhide** [comment]
Awesome, thanks @DunyaKokoschka! ❤️ 

Would love a follow-up PR that adds those sweet `@tag :skip`s if you're feeling like it 😄
